### PR TITLE
K8s set of examples

### PR DESCRIPTION
**Pull request by ctr26**
Attached is an example script that runs as a single job to demonstrate the k8s python api. Included are examples for job creation delete updating and creating jobs from ymls.

N.B. there needs to be a specific reference to namespaces else jobs do get dumped into default by (unsurprisingly) default

### CODE REVIEW

**Comment by ctr26** [comment]
I can resubmit this pull request to it's own branch if you'd rather

File: k8s-api/k8s-api.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+def main():
+
+    # Configs can be set in Configuration class directly or using helper
+    # utility. If no argument provided, the config will be loaded from
+    # default location.
+
+    try:
+        config.load_kube_config()
+    except:
```
Comment (Alexey-ebi) [inline k8s-api/k8s-api.py:67]: It should be ConfigException

File: k8s-api/k8s-api.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
+from kubernetes import client, config, utils
+from os import path
+
+JOB_NAME = "pi"
+NAMEPACE = "covid"
```
Comment (Alexey-ebi) [inline k8s-api/k8s-api.py:5]: should be NAMESPACE?

File: k8s-api/k8s-api.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    c.verify_ssl = False  # set verify_ssl to false in that config
+    # make that config the default for all new clients
+    client.Configuration.set_default(c)
+    v1 = client.CoreV1Api()
+    api = client.ApiClient()
+
+    try:
+        current_namespace = open(
+            "/var/run/secrets/kubernetes.io/serviceaccount/namespace").read()
+    except:
```
Comment (Alexey-ebi) [inline k8s-api/k8s-api.py:81]: Exception is too broad, what type of Exception do you expect here?

File: k8s-api/k8s-api.py [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    # default location.
+
+    try:
+        config.load_kube_config()
+    except:
+        # load_kube_config throws if there is no config, but does not document what it throws, so I can't rely on any particular type here
+        config.load_incluster_config()
+
+    c = client.Configuration()  # go and get a copy of the default config
+    c.verify_ssl = False  # set verify_ssl to false in that config
```
Comment (Alexey-ebi) [inline k8s-api/k8s-api.py:72]: Do we need to do this? I got this warning when tried to list pods: InsecureRequestWarning: Unverified HTTPS request is being made to host '10.43.0.1'. Adding certificate verification is strongly advised.

File: k8s-api/k8s-api.py [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+    job = client.V1Job(
+        api_version="batch/v1",
+        kind="Job",
+        metadata=client.V1ObjectMeta(name=JOB_NAME),
+        spec=spec)
+
+    return job
+
+
+def create_job(api_instance, job):
```
Comment (Alexey-ebi) [inline k8s-api/k8s-api.py:32]: Please create docstring here

File: k8s-api/k8s-api.py [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+    api_response = api_instance.create_namespaced_job(
+        body=job,
+        namespace=NAMEPACE)
+    print("Job created. status='%s'" % str(api_response.status))
+
+
+def update_job(api_instance, job):
```
Comment (Alexey-ebi) [inline k8s-api/k8s-api.py:39]: docstring required, need to describe parameters of a function

File: k8s-api/k8s-api.py [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+    # Update container image
+    job.spec.template.spec.containers[0].image = "perl"
+    api_response = api_instance.patch_namespaced_job(
+        name=JOB_NAME,
+        namespace=NAMEPACE,
+        body=job)
+    print("Job updated. status='%s'" % str(api_response.status))
+
+
+def delete_job(api_instance,job_name,namespace):
```
Comment (Alexey-ebi) [inline k8s-api/k8s-api.py:49]: docstring required, need to describe parameters of a function

**Pull request closed without merging**